optimize/backend/src/main/java/io/camunda/optimize/service/db/os/writer/InstantDashboardMetadataWriterOS.java

```diff
@@ -69,38 +70,53 @@ public void saveInstantDashboard(final InstantDashboardDataDto dashboardDataDto)
-
-    final BulkRequest.Builder bulkRequestBuilder = new BulkRequest.Builder();
+    final List<String> dashboardIdsToBeDeleted = new ArrayList<>();
     final SearchRequest.Builder requestBuilder =
         new SearchRequest.Builder()
             .index(INSTANT_DASHBOARD_INDEX_NAME)
-            .query(not(longTerms(InstantDashboardDataDto.Fields.templateHash, hashesAllowed)));
+            .query(
+                q ->
+                    q.bool(
+                        b ->
+                            b.mustNot(
+                                m ->
+                                    m.terms(
+                                        t ->
+                                            t.field(InstantDashboardDataDto.Fields.templateHash)
+                                                .terms(
+                                                    tt ->
+                                                        tt.value(
+                                                            hashesAllowed.stream()
+                                                                .map(FieldValue::of)
+                                                                .toList()))))));
+
+    final SearchResponse<InstantDashboardDataDto> searchResponse =
+        osClient.search(
+            requestBuilder,
+            InstantDashboardDataDto.class,
+            "Some errors occurred while deleting outdated instant dashboards.");
 
-    final List<String> dashboardIdsToBeDeleted =
-        osClient.searchValues(requestBuilder, Result.class).stream()
-            .map(Result::dashboardId)
-            .toList();
-
-    if (!dashboardIdsToBeDeleted.isEmpty()) {
-      dashboardIdsToBeDeleted.forEach(
-          id ->
+    final BulkRequest.Builder bulkRequestBuilder = new BulkRequest.Builder();
+    LOG.debug(
+        "Deleting [{}] instant dashboard documents by id with bulk request.",
+        searchResponse.hits().hits().size());
+    searchResponse
+        .hits()
+        .hits()
+        .forEach(
+            hit -> {
+              assert hit.source() != null;
+              dashboardIdsToBeDeleted.add(hit.source().getDashboardId());
               bulkRequestBuilder.operations(
-                  op -> op.delete(del -> del.index(INSTANT_DASHBOARD_INDEX_NAME).id(id))));
-
-      LOG.debug(
-          "Deleting [{}] instant dashboard documents by id with bulk request.",
-          dashboardIdsToBeDeleted.size());
-
-      osClient.bulk(
-          bulkRequestBuilder,
-          format(
-              "Failed to bulk delete from %s %s outdated template entries: %s",
-              INSTANT_DASHBOARD_INDEX_NAME,
-              dashboardIdsToBeDeleted.size(),
-              dashboardIdsToBeDeleted));
-    }
-
+                  o ->
+                      o.delete(
+                          OptimizeDeleteOperationBuilderOS.of(
+                              d ->
+                                  d.optimizeIndex(osClient, INSTANT_DASHBOARD_INDEX_NAME)
```

Comment:
Same here - the change looks a bit an overkill for just adding index prefix, but should now be aligned with ES impl, I suppose

Reply:
@oleksandr-kriuchenko-lohika The issue was deeper. It was not just about the prefix. The code was looking for the IDs of dashboards, however it should have been looking for the IDs of the containers of those dashboards (instant preview dashboard), therefore this method was actually deleting the wrong thing